Update Program2.java

Mel98378 opened the pull request:
added get_valid_output function

Mel98378 committed:
Update Program2.java

added get_valid_output function

lavender-aa commented on program-2/Program2.java:
```diff
@@ -127,7 +127,81 @@ static String get_file_name(Data data, BufferedReader reader) {
+                out_name = args[1];
+            } else {
+                System.out.print("Enter output file name: ");
+                out_name = get_file_name(data, reader);
```
not sure how or why, but by the time `reader.readLine()` gets called inside `get_file_name()`, the passed buffer is closed

lavender-aa replied:
removing the stream closing statement inside the get_valid_input seems to fix it?? i have no idea what's going on

lavender-aa replied:
how to fix: at the end of `get_valid_input()` , there is a try/catch block pertaining to closing the buffered reader. wrap it inside an if statement that only runs if quit is true

lavender-aa commented on program-2/Program2.java:
```diff
@@ -127,7 +127,81 @@ static String get_file_name(Data data, BufferedReader reader) {
+                                    System.out.println("Failed to create a backup. Proceeding with file overwrite.");
+                                    // fall through to overwrite
+                                }
+                                break;
```
this case will never fall through to the below case (due to the `break`)

lavender-aa commented on program-2/Program2.java:
```diff
@@ -127,7 +127,81 @@ static String get_file_name(Data data, BufferedReader reader) {
+                            case "3":
+                                // proceed to overwrite the file
+                                System.out.println("Overwriting the existing output file.");
+                                break;
```
this option does nothing -- it will return the user to the "output file already exists" prompt

lavender-aa commented on program-2/Program2.java:
```diff
@@ -127,7 +127,81 @@ static String get_file_name(Data data, BufferedReader reader) {
+                                    System.out.println("Backup of existing file created as " + backup.getName());
+                                    // proceed to use the original file as output
+                                    System.out.print("Enter new output file name: ");
+                                    out_name = get_file_name(data, reader);
```
don't make the user enter in the name they just said they wanted to use -- if a backup of a file is made, use the original file name as the output

Mel98378 committed:
Update Program2.java

added if statement around try/catch in get_file_name function and removed the re-enter name prompt.

lavender-aa commented on program-2/Program2.java:
```diff
@@ -111,14 +111,16 @@ static String get_file_name(Data data, BufferedReader reader) {
+                        data.set_quit(true);
+                    }
+                } catch (IOException e) {
+                    System.out.println("Error reading input file name.");
```
this is not what i described in my previous comment. please read it carefully

lavender-aa replied:
(to be clear, revert this change. not only does it not solve the original problem, it introduces a compiler bug.)

Mel98378 committed:
Update Program2.java

removed if statement from get_file_name and added if statement to get_valid_input

lavender-aa commented:
if you update your branch you should be able to test more easily.

if you forgot how (in GHD):
- switch to main branch
- sync/fetch/pull (button in top right)
- switch to your branch
- click branches tab, click "select a branch to merge into Camron" (at bottom)
- select main, merge

lavender-aa committed:
merge main

lavender-aa committed:
removed extra level of indent from methods inside `Program2`

lavender-aa committed:
fixed get_valid_output(), adjusted word printing loop to work properly

output file options actually work now that they can stop the while loop. word printing loop needed to be adjusted because having just 1 word would have last index at 0 and i starting at 0 being strictly less than last index, so the word wouldn't get printed to the output. i have confirmed that there is no index out of bounds and that all 100 words do actually print

lavender-aa commented:
decided to do this while i'm still awake and not sleepy and groggy in the morning. updated the branch with main, fixed it, tested it a decent bit. you guys can feel free to test the main branch in the morning if you get the time to, but i don't think there are any obvious errors.

i have gone over the rubric several times, and as it is currently it should cover almost, if not, every item on the rubric
